app/models/stage.rb

```diff
@@ -28,6 +28,8 @@ class Stage < ActiveRecord::Base
   attr_writer :command
   before_save :build_new_project_command
 
+  before_soft_delete :verify_not_part_of_pipeline
```

Comment:
Yea, unfortunately they made the decision to load the plugin classes before the main app classes, thus you'll see the ```prepend``` directive scattered about to work around this.